addon/mixins/scoped-query-cache.js

```diff
@@ -73,5 +76,21 @@ export default Ember.Mixin.create({
 
       return results;
     });
+  },
+
+  scheduleCacheDecay(delay = CACHE_DECAY_TIMEOUT, callback) {
+    this.decayId = setTimeout(() => {
+      this.cache = this._createCache();
+
+      callback && typeof callback === 'function' && callback();
```

Comment:
Unclear if we need this. I added specifically for testing, but need to think through the use cases to determine whether it's required.